backend/main.py

```
@@ -136,6 +166,107 @@ async def delete_product(id: int, db: AsyncSession = Depends(get_db)):
+
+@app.get("/cart/", response_model=List[CartItemResponse])
+async def get_cart_items(db: AsyncSession = Depends(get_db)):
+    result = await db.execute(
+        select(CartItem).options(selectinload(CartItem.product))
+    )
+    cart_items = result.scalars().unique().all()
+    return cart_items
+
+
+@app.delete("/cart/{id}")
+async def remove_from_cart(id: int, db: AsyncSession = Depends(get_db)):
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)